### ISSUE TITLE
abaplint does not catch syntax error (char->string assignment)

### ISSUE THREAD

**Issue by FreHu** [abaplint/abaplint#2185]
Here, iv_transport has the domain trkorr (char20), it is a syntax error to pass it to a string parameter
https://github.com/abapGit/abapGit/pull/5022/commits/483dcff5c5800d2ebd334c33b52c1d3638bbc737

I feel like abap sometimes allows it and sometimes doesn't. I've also seen function modules dump when passed a string parameter and they expect some character domain.

**Comment by FreHu** [comment]
```abap
REPORT zfh3.

TYPES: t_shorter_char TYPE c LENGTH 10.
TYPES: t_longer_char TYPE c LENGTH 20.
TYPES: t_string TYPE string.

DATA: short TYPE t_shorter_char.
DATA: long TYPE t_longer_char.
DATA: string TYPE t_string.

" there are no syntax errors during regular assignment
short = string. " value might be cut off
short = long. " value might be cut off
short = short. " maybe a different issue, but this doesn't do anything

long = short. " ok
long = string. " same as short = string, a value can be cut off if it has fixed length

string = short. " ok
string = long. " ok

CLASS lcl_test DEFINITION.

  PUBLIC SECTION.
    CLASS-METHODS in
      IMPORTING short  TYPE t_shorter_char OPTIONAL
                long   TYPE t_longer_char OPTIONAL
                string TYPE t_string OPTIONAL.

  PROTECTED SECTION.
  PRIVATE SECTION.

ENDCLASS.

CLASS lcl_test IMPLEMENTATION.

  METHOD in.
  ENDMETHOD.

ENDCLASS.

START-OF-SELECTION.

  lcl_test=>in( short = short ).
  lcl_test=>in( short = long ).
  lcl_test=>in( short = string ).
  
  lcl_test=>in( long = short ).
  lcl_test=>in( long = long ).
  lcl_test=>in( long = string ).
  
  lcl_test=>in( string = short ).
  lcl_test=>in( string = long ).
  lcl_test=>in( string = string ).
  ```
![image](https://user-images.githubusercontent.com/5097067/136263382-34c9ea3a-96a7-49bb-8832-7a447bebf949.png)
